python/pyspark/sql/streaming/list_state_client.py

```diff
@@ -78,8 +78,11 @@ def get(self, state_name: str, iterator_id: str) -> Tuple:
             status = response_message[0]
             if status == 0:
                 iterator = self._stateful_processor_api_client._read_arrow_state()
-                batch = next(iterator)
-                pandas_df = batch.to_pandas()
+                data_batch = None
```

Comment:
A bit confused here, this code snippet is trying to deal with multiple batches but only keep the data in the last batch?

Reply:
The previous code would stuck forever after we added the arrow resource cleanup logic (I think it might be related to previous logic did not exhaust the iterator, though that iterator did only contain a single batch), hence using the recommended way to consume the arrow batches which is
```
for batch in iterator:
    ......
```
The logic is the same as the previous one, we only need to consume a single batch here.

Reply:
Can we have short comment here to explain the iteration though we are not expecting multiple batches?

Reply:
It's OK to have the fix in here, as the fix is applied to MapState impl as well.